software/base/src/main/java/org/apache/brooklyn/entity/software/base/VanillaWindowsProcessImpl.java

```diff
@@ -29,7 +33,12 @@ public Class getDriverInterface() {
     protected void preStart() {
         super.preStart();
         sensors().set(RDP_PORT, 3389);
-        sensors().set(WINRM_PORT, 5985);
+        Location loc = Iterables.getFirst(getLocations(), null);
+        if (loc != null) {
+            if(loc.getConfig(WinRmMachineLocation.USE_SECURE_WINRM)) {
+                sensors().set(WINRM_PORT, 5986);
+            }
+        }
```

Comment:
Why not handle the  `!USE_SECURE_WINRM` case? Should be possible to use both protocols.